app/src/main/java/com/github/libretube/helpers/ProxyHelper.kt

```diff
@@ -27,4 +27,24 @@ object ProxyHelper {
             ?.port(proxyUrl.port)
             ?.toString()
     }
+
+    /**
+     * Load the YT url directly instead of the proxied Piped URL if enabled
+     */
+    fun unwrapIfEnabled(url: String): String {
+        if (!PreferenceHelper.getBoolean(
+                PreferenceKeys.DISABLE_VIDEO_IMAGE_PROXY,
+                false
+            )
+        ) {
+            return url
+        }
+
+        return url.toHttpUrlOrNull()?.let {
+            it.newBuilder()
+                .host(it.queryParameter("host").orEmpty())
+                .removeAllQueryParameters("host")
+                .build()
+        }.toString()
```

Comment:
```suggestion
        return url.toHttpUrlOrNull()?.let {
            it.newBuilder()
                .host(it.queryParameter("host").orEmpty())
                .removeAllQueryParameters("host")
                .build()
                .toString()
        }
```

This avoids returning a "null" string if the URL is invalid.